src/request/src/cache.js

```diff
@@ -70,9 +68,9 @@ export default class CacheRequest extends Request {
 
   set url(urlString) {
     super.url = urlString;
-    let pathname = global.decodeURIComponent(url.parse(urlString).pathname);
-    const urlParts = pathname.replace(/^\//g, '').split('/');
-    // "pathname" has the following form "/namespace/appKey/collection/id"
+    const pathname = global.decodeURIComponent(url.parse(urlString).pathname);
```

Comment:
We did observe some encoding done by parse(), but it wasn't in the docs so I'm confused. And double encoding (once in the core sdk and once in the NativeScript SDK) was the cause of a misrepresented auth token and 401 when trying to find() a file after uploading. So I think we should look into this and make a decision when how and where URLs are encoded and reach a "single point of truth" about it. Just not sure where :) I think somewhere in the core SDK...